core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -891,10 +891,12 @@ private[spark] object SparkSubmitUtils {
     val localIvy = new FileSystemResolver
     val localIvyRoot = new File(ivySettings.getDefaultIvyUserDir, "local")
     localIvy.setLocal(true)
-    localIvy.setRepository(new FileRepository(localIvyRoot))
```

Comment:
Everything else being equal, I'm in favor of leaving this. Maybe it's redundant but it might prevent bugs in case someone else comes along and tries to remove the use of absolute paths below (or forgets to use an absolute path in new code).